### ISSUE TITLE
error using omeg-supreme method

### ISSUE THREAD

**Issue by siddo420** [primus/metroplex#13]
I have added metroplex with omega-supreme.
```js
  primus.use( 'omega-supreme',    require( 'omega-supreme'    )); // before metroplex
  primus.use( 'metroplex',        require( 'metroplex'        )); // registry/map 
```

Then, I have the following code snippet
``` js
primus.forward.sparks( sparks2send, 
				      msg,
				     function( err, results ){
		if( err ) {
		  Logger.errorc( 'Error: ' + err  );
		  return done( err, null );
                }
//  ........
});
```
The code above prints following error:
```js
Error: Error: Invalid URI "/primus/omega/supreme"
```

Example list of sparks:
```js
sparks2send = ["64xRm74gJ2dsMXVJAAAA","5iQsCn0grQ5EImWMAAAB","LBD9Vtykb5vo5qWrAAAB","UoA4Ir54PP9sCO-qAAAB","mlNYdagAleP1AzuJAAAA","xoc7pJzICFj1IlINAAAB","ffSiCA3n6HGVaUO5AAAA","Wi744PWyRCbmp5OfAAAI","nbIvK_GwjSLKGihtAAAA","eE5FnKSTBxNryBRXAAAA","BfadUHua5RZ3HVqEAAAA","ySd-xtbqgHo4gLNmAAAA","W5ioExC5VRKwcxB0AAAA","vnKVluWrieQP0jCSAAAA"]
```
I may be missing something in configuration I think:
```js
  primus_options =   {  
        pathname:     '/primus'   // The URL namespace that Primus can own
      , parser:       'JSON'
      , transformer:  'engine.io'     
      , namespace:    'mp'        // for 'mp'=metroplex prefixed keys
      , address:      add
      , username:     'user'  
      , password:     'password'  
      , redis: Redis.getClient()
      , concurrently: 30 //How many servers can we broadcast to at once, defaults to 10
      , url: '/primus/omega/supreme'  //-->doesn't work with/without this option
   };
  
```

**Comment by lpinca** [comment]
There was a bug that I've just fixed in 1afdc8 but it was unrelated to the issue you are reporting.
I was unable to reproduce your issue. In this [gist](https://gist.github.com/lpinca/c1b92a9d4ebae5b34574) you can find the test case I've used.

**Comment by siddo420** [comment]
But, your test fails if an invalid/non-existent spark is given to primus.forward.sparks().

I made the following change to your server code and ran it again:
```
    sparks.push(spark.id);
    if (sparks.length !== 2) return;
    sparks.push('234fwfrewrw423#%#$'); //<---- added this line
```
Here is what happens as a result:
```
server listening on 0.0.0.0:3002
server listening on 0.0.0.0:3001
L1mXUWv connected
L1mXUX9 connected
/private/tmp/primus_test/node_modules/metroplex/node_modules/ioredis/node_modules/bluebird/js/main/async.js:43
        fn = function () { throw arg; };
                           ^
Error: Invalid URI "/primus/omega/supreme"   <-- same error
    at Request.init (/private/tmp/primus_test/node_modules/omega-supreme/node_modules/request/request.js:289:31)
```

This is a valid case because sparks can be disconnected anytime (even during the call to .forwards() method).

Also, if you decide to fix it, please make sure you return an array of results with one result item for each corresponding spark in the input array like you do for metroplex.sparks().

**Comment by siddo420** [comment]
['L1mXUWv', 'L1mXUX9', ''234fwfrewrw423#%#$'']

So, if above is the input list of sparks and assuming that 3rd item is invalid, the output should be similar to:

[ 'ok','ok',null ]

**Comment by lpinca** [comment]
It should be fixed in master.

> Also, if you decide to fix it, please make sure you return an array of results with one result item for each corresponding spark in the input array like you do for metroplex.sparks().

Unfortunately this is not possible as the return value is the [info object](https://github.com/primus/omega-supreme#messaging) returned by `omega-supreme`.

**Comment by siddo420** [comment]
Should I create a new issue at omega-supreme?

This is important for client, so it could record the info in DB if the pkt was sent to a spark successfully or not (for re-attempts etc).

**Comment by lpinca** [comment]
Yes, do it.

**Comment by siddo420** [comment]
Done.
https://github.com/primus/omega-supreme/issues/5

**Comment by siddo420** [comment]
was https://github.com/primus/metroplex/commit/780c2358a1cb7e4cebd79192a168b0410437ad2f not added to the latest release?

**Comment by lpinca** [comment]
nope.

**Comment by siddo420** [comment]
when will it be done? And, what are we waiting for? I mean is it safe to get it from the repo directly?

**Comment by lpinca** [comment]
The problem is that if we publish current master we will force people to use https://github.com/luin/ioredis.
If @3rd-Eden is ok I'll go ahead and publish `metroplex@1.0.0`.

You can safely install from github in the meanwhile.

**Comment by 3rd-Eden** [comment]
@lpinca go for it.

**Comment by lpinca** [comment]
Published `metroplex@1.0.0`.
